addons/html_builder/static/src/builder/plugins/builder_overlay/builder_overlay_plugin.js

```diff
@@ -1,34 +1,234 @@
 import { Plugin } from "@html_editor/plugin";
-import { BuilderOverlay } from "./builder_overlay";
+import { throttleForAnimation } from "@web/core/utils/timing";
+import { renderToElement } from "@web/core/utils/render";
+import { getScrollingElement, getScrollingTarget } from "@web/core/utils/scrolling";
+import { isMobileView } from "../../utils/utils";
 
 export class BuilderOverlayPlugin extends Plugin {
-    static id = "builder_overlay";
-    static dependencies = ["selection", "overlay"];
-    static resources = (p) => ({
-        change_selected_toolboxes_listeners: p.openBuilderOverlay.bind(p),
-    });
+    static id = "builderOverlay";
+    static dependencies = ["selection", "localOverlay"];
+    resources = {
+        step_added_handlers: this._update.bind(this),
+        change_selected_toolboxes_listeners: this.openBuilderOverlay.bind(this),
+    };
 
     setup() {
-        this.overlay = this.dependencies.overlay.createOverlay(BuilderOverlay, {
-            positionOptions: {
-                position: "center",
-            },
+        this.overlayContainer = this.dependencies.localOverlay.makeLocalOverlay(
+            "builder-overlay-container"
+        );
+        this.overlays = [];
+        this.toolboxes = [];
+
+        this.update = throttleForAnimation(this._update.bind(this));
+
+        // Recompute the overlay when the window is resized.
+        this.addDomListener(window, "resize", this.update);
+
+        // On keydown, hide the overlay and then show it again when the mouse
+        // moves.
+        let wasKeydown;
```

Comment:
I would rather have a variable `isOverlayVisible` set by `toggleOverlayVisibility`. If we try to set a visibility that is the same as the previous one, we directly return.